src/snowflake/cli/api/rendering/jinja.py

```diff
@@ -82,8 +82,37 @@ def getitem(self, obj, argument):
             return self.undefined(obj=obj, name=argument)
 
 
+def _get_jinja_env(loader: Optional[loaders.BaseLoader] = None) -> Environment:
+    return env_bootstrap(
+        IgnoreAttrEnvironment(
+            loader=loader or loaders.BaseLoader(),
+            keep_trailing_newline=True,
+            undefined=StrictUndefined,
+        )
+    )
+
+
+def jinja_render_from_str(template_content: str, data: Dict[str, Any]) -> str:
+    """
+    Renders a jinja template and outputs either the rendered contents as string or writes to a file.
+
+    Args:
+        template_content (str): template contents
+        data (dict): A dictionary of jinja variables and their actual values
+
+    Returns:
+        None if file path is provided, else returns the rendered string.
+    """
+    return _get_jinja_env().from_string(template_content).render(data)
+
+
+def path_to_jinja_pathlike_str(path: Path) -> str:
+    # jinja2 template loader user '/' as path separator (even on Windows)
+    return "/".join(path.parts)
```

Comment:
have you tried path.to_posix()? It seems to work well for me.

Reply:
Nice one!